scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/GraphQLQueries.java

```diff
@@ -147,60 +147,179 @@ private JobInput getJobInput(List<String> statusNames, String user, List<Constra
                 }
                 case USER: {
-                    user = getFilteringString(constraint.getAction(), user, value);
+
+                    switch (constraint.getAction()) {
+                        case EQUALS:
+                            if (value != null) {
+                                if (userFilter == null)
+                                    userFilter = value;
+                                else if (value != userFilter)
+                                    return input.jobName(RETURN_NOTHING_FILTER).build();
+                            }
+                            break;
+                        case NOT_EQUAL:
+                            if (value != null)
+                                user = "!" + value;
+                            break;
+                        case CONTAINS:
+                            if (value != null)
+                                user = "*" + value + "*";
+                            break;
+                        case NOT_CONTAIN:
+                            if (value != null)
+                                user = "!*" + value + "*";
+                            break;
+                        case STARTS_WITH:
+                            if (value != null &&
+                                (user == null || (value.startsWith(user) && value.length() > user.length())))
+                                user = value + "*";
+                            break;
+                        default:
+                            break;
+                    }
                     break;
                 }
                 case NAME: {
-                    name = getFilteringString(constraint.getAction(), name, value);
+
+                    switch (constraint.getAction()) {
+                        case EQUALS:
+                            if (value != null) {
+                                if (name == null)
+                                    name = value;
+                                else if (value != name)
+                                    return input.jobName(RETURN_NOTHING_FILTER).build();
+                            }
+                            break;
+                        case NOT_EQUAL:
+                            if (value != null)
+                                name = "!" + value;
+                            break;
+                        case CONTAINS:
+                            if (value != null)
+                                name = "*" + value + "*";
+                            break;
+                        case NOT_CONTAIN:
+                            if (value != null)
+                                name = "!*" + value + "*";
+                            break;
+                        case STARTS_WITH:
+                            if (value != null &&
+                                (name == null || (value.startsWith(name) && value.length() > name.length())))
+                                name = value + "*";
```

Comment:
the else of these conditions does nothing?

Reply:
Exactly. But see refactoring in the last commit.

Reply:
Ok, great !